data_broker.js

```diff
@@ -199,7 +210,7 @@ class DataBroker extends EventTarget {
   }
 
   async _updateStats() {
-    const validRegions = this.regions ? this.regions : getValidRefs(this._header, this._readDepthData);
+    const validRegions = getValidRefs(this._header, this._readDepthData);
```

Comment:
Here I changed it because the region won't be updated if loading another file. 

E.g. When you open the demo file first, then load another file on the home page, the header and readDepth will be changed. The regions should be updated as well. However, I found that the regions were still the prior one.

Reply:
But it's completely broken sampling. If you select a chromosome and look at the `/alignmentStatsStream` request you can see that it's sampling all chromosomes instead of just the one selected. Try to be careful and make sure you understand what code is doing before you change it. If I hadn't caught this in review (I miss things a lot) it would have introduced a fairly serious bug, because it appears to be working correctly but it's not actually showing the user the data they think it is.

Reply:
Sorry, I saw the line of the code `let allRegions = validRegions;` I thought it was the all regions. Are you saying that the `this.region` is either single region or the `all regions`? 

I'll double check the return from `/alignmentStatsStream`.

Reply:
It depends on context. "all regions" might mean different things depending on what part of the code you are in. I don't remember specifically what it means there. It might need a better name in that function. You can look in index.html to see where this.regions is set (broker.regions).